id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java

```diff
@@ -156,6 +155,62 @@ public ResponseWrapper<CredentialIssueResponse> createCredentialIssuance(Credent
 		return credentialIssueResponseWrapper;
 	}
 
+	@Override
+	public ResponseWrapper<CredentialIssueResponse> createCredentialIssuanceByRid(CredentialIssueRequestDto credentialIssueRequestDto, String rid) {
+		LOGGER.debug(IdRepoSecurityManager.getUser(), CREDENTIAL_SERVICE, CREATE_CREDENTIAL,
+				"started creating credential");
+		List<ServiceError> errorList = new ArrayList<>();
+		ResponseWrapper<CredentialIssueResponse> credentialIssueResponseWrapper = new ResponseWrapper<CredentialIssueResponse>();
+
+		CredentialIssueResponse credentialIssueResponse = null;
+		try{
+			CredentialEntity credential=new CredentialEntity();
+			credential.setRequestId(rid);
+			credential.setRequest(mapper.writeValueAsString(credentialIssueRequestDto));
+			credential.setStatusCode(CredentialStatusCode.NEW.name());
+			credential.setCreateDateTime(LocalDateTime.now(ZoneId.of("UTC")));
+			credential.setCreatedBy(IdRepoSecurityManager.getUser());
+			credential.setStatusComment("Request created");
+			credentialDao.save(credential);
+			credentialIssueResponse = new CredentialIssueResponse();
+			credentialIssueResponse.setRequestId(rid);
+			credentialIssueResponse.setId(credentialIssueRequestDto.getId());
+			LOGGER.debug(IdRepoSecurityManager.getUser(), CREDENTIAL_SERVICE, CREATE_CREDENTIAL,
+					"ended creating credential");
+		}catch(DataAccessLayerException e) {
+			auditHelper.auditError(AuditModules.ID_REPO_CREDENTIAL_REQUEST_GENERATOR,
+					AuditEvents.CREATING_CREDENTIAL_REQUEST, rid, IdType.ID, e);
+			ServiceError error = new ServiceError();
+			error.setErrorCode(CredentialRequestErrorCodes.DATA_ACCESS_LAYER_EXCEPTION.getErrorCode());
+			error.setMessage(CredentialRequestErrorCodes.DATA_ACCESS_LAYER_EXCEPTION.getErrorMessage());
+			errorList.add(error);
+			LOGGER.error(IdRepoSecurityManager.getUser(), CREDENTIAL_SERVICE, CREATE_CREDENTIAL, ExceptionUtils.getStackTrace(e));
+		} catch (Exception e) {
+			auditHelper.auditError(AuditModules.ID_REPO_CREDENTIAL_REQUEST_GENERATOR,
+					AuditEvents.CREATING_CREDENTIAL_REQUEST, rid, IdType.ID, e);
+			ServiceError error = new ServiceError();
+			error.setErrorCode(CredentialRequestErrorCodes.UNKNOWN_EXCEPTION.getErrorCode());
+			error.setMessage(CredentialRequestErrorCodes.UNKNOWN_EXCEPTION.getErrorMessage());
+			errorList.add(error);
+			LOGGER.error(IdRepoSecurityManager.getUser(), CREDENTIAL_SERVICE, CREATE_CREDENTIAL, ExceptionUtils.getStackTrace(e));
+		} finally {
+			credentialIssueResponseWrapper.setId(EnvUtil.getCredReqServiceId());
+			DateTimeFormatter format = DateTimeFormatter.ofPattern(EnvUtil.getDateTimePattern());

```

Comment:
You can use io.mosip.kernel.core.util.DateUtils.getUTCCurrentDateTime(), no need for getting a formatter and format the date time again back to LocalDateTime.

Reply:
taken care